src/blas/backends/cublas/cublas_level1.cpp

```diff
@@ -0,0 +1,604 @@
+    // THis requires to bring the data to host, copy it, and return it back to
+    // the device
+
+    result.template get_access<cl::sycl::access::mode::write>()[0] =
+        std::max((int64_t)int_res_buff.template get_access<cl::sycl::access::mode::read>()[0] - 1,
+                 int64_t{ 0 });
+}
+
+#define IAMAX_LAUNCHER(TYPE, CUBLAS_ROUTINE)                                    \
+    void iamax(cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<TYPE, 1> &x, \
+               const int64_t incx, cl::sycl::buffer<int64_t, 1> &result) {      \
+        iamax(CUBLAS_ROUTINE, queue, n, x, incx, result);                       \
+    }
+IAMAX_LAUNCHER(float, cublasIsamax)
+IAMAX_LAUNCHER(double, cublasIdamax)
+IAMAX_LAUNCHER(std::complex<float>, cublasIcamax)
+IAMAX_LAUNCHER(std::complex<double>, cublasIzamax)
+#undef IAMAX_LAUNCHER
+
+template <typename Func, typename T>
+inline void swap(Func func, cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<T, 1> &x,
+                 int64_t incx, cl::sycl::buffer<T, 1> &y, int64_t incy) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+    overflow_check(n, incx, incy);
+    queue.submit([&](cl::sycl::handler &cgh) {
+        auto x_acc = x.template get_access<cl::sycl::access::mode::read_write>(cgh);
+        auto y_acc = y.template get_access<cl::sycl::access::mode::read_write>(cgh);
+        cgh.interop_task([=](cl::sycl::interop_handler ih) {
+            auto sc     = CublasScopedContextHandler(queue);
+            auto handle = sc.get_handle(queue);
+            auto x_     = sc.get_mem<cuDataType *>(ih, x_acc);
+            auto y_     = sc.get_mem<cuDataType *>(ih, y_acc);
+            cublasStatus_t err;
+            CUBLAS_ERROR_FUNC(func, err, handle, n, x_, incx, y_, incy);
+        });
+    });
+}
+
+#define SWAP_LAUNCHER(TYPE, CUBLAS_ROUTINE)                                                  \
+    void swap(cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<TYPE, 1> &x, int64_t incx, \
+              cl::sycl::buffer<TYPE, 1> &y, int64_t incy) {                                  \
+        swap(CUBLAS_ROUTINE, queue, n, x, incx, y, incy);                                    \
+    }
+
+SWAP_LAUNCHER(float, cublasSswap)
+SWAP_LAUNCHER(double, cublasDswap)
+SWAP_LAUNCHER(std::complex<float>, cublasCswap)
+SWAP_LAUNCHER(std::complex<double>, cublasZswap)
+#undef SWAP_LAUNCHER
+
+template <typename Func, typename T>
+inline void iamin(Func func, cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<T, 1> &x,
+                  const int64_t incx, cl::sycl::buffer<int64_t, 1> &result) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+    overflow_check(n, incx);
+    // cuda does not support int64_t as return type for the data. So we need to
+    // mimic iamax
+    // we are converting the result to be the int and then we convert it back to
+    // the actual data on the host
+    // FIXME:: this change may cause failiour as the result of integer overflow
```

Comment:
FIXME? Is this supposed to be fixed already? Also, there are typos in the comments.

Reply:
Yes, the comment was outdated, I have updated the comments.